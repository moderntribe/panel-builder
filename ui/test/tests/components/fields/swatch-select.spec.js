/* global expect b:true */

import React from 'react';
import { shallow } from 'enzyme';
import SwatchSelect from 'components/fields/swatch-select';
import renderer from 'react-test-renderer';

describe('SwatchSelect field', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(<SwatchSelect description="This is a test description" label="Some test label" />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders as a <div>', () => {
		const wrapper = shallow(<SwatchSelect />);
		expect(wrapper.type()).toEqual('div');
	});


	it('has no color select radio buttons without properties.', () => {
		const wrapper = shallow(<SwatchSelect />);
		expect(wrapper.find('input').length).toEqual(0);
	});

	it('has a label', () => {
		const wrapper = shallow(<SwatchSelect />);
		expect(wrapper.find('label').length).toEqual(1);
	});
});
