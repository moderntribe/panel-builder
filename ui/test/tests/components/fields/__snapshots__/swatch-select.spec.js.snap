exports[`SwatchSelect field renders correctly 1`] = `
<div
  className="field panel-field panel-conditional-field">
  <label
    className="label panel-field-label">
    Some test label
    <span
      className="icon"
      onBlur={undefined}
      onClick={undefined}
      onContextMenu={undefined}
      onFocus={undefined}
      onMouseDown={undefined}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={undefined} />
  </label>
  <div
    className="container" />
</div>
`;
